nativelib/src/main/resources/gc/commix/Heap.h

```diff
@@ -0,0 +1,99 @@
+#ifndef IMMIX_HEAP_H
+#define IMMIX_HEAP_H
+
+#include "GCTypes.h"
+#include "Allocator.h"
+#include "LargeAllocator.h"
+#include "datastructures/Bytemap.h"
+#include "datastructures/BlockRange.h"
+#include "datastructures/GreyPacket.h"
+#include "metadata/LineMeta.h"
+#include "Stats.h"
+#include <stdio.h>
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <pthread.h>
+#include <semaphore.h>
+
+typedef enum {
+    gc_idle = 0x0,
+    gc_mark = 0x1,
+    gc_sweep = 0x2
+} GCThreadPhase;
+
+typedef struct {
+    word_t *blockMetaStart;
+    word_t *blockMetaEnd;
+    word_t *lineMetaStart;
+    word_t *lineMetaEnd;
+    word_t *heapStart;
+    word_t *heapEnd;
+    word_t *greyPacketsStart;
+    size_t heapSize;
+    size_t maxHeapSize;
+    uint32_t blockCount;
+    uint32_t maxBlockCount;
+    struct {
+        sem_t start;
+        sem_t start0;
```

Comment:
What's the difference between `start` and `start0` ?

Reply:
`start0` is for GC Thread 0 (main GC thread)
`start` is for all other GC threads
I will change all the suffixes 0 to `Main` or `Master` so it is more explicit.